config/routes.rb

```diff
@@ -75,311 +75,332 @@
+        resources :environments do
+          scope module: 'environments/relationships' do
+            resources :tokens, only: %i[index show create]
+          end
         end
+
+        resources :event_logs, path: 'event-logs', only: %i[index show]
       end
-    end
 
-    ee do
-      resources :request_logs, path: 'request-logs', only: %i[index show]  do
+      resources :metrics, only: %i[index show] do
         collection do
-          scope :actions, module: 'request_logs/actions' do
+          scope :actions, module: 'metrics/actions' do
             get :count, to: 'counts#count'
           end
         end
       end
 
-      resources :environments do
-        scope module: 'environments/relationships' do
-          resources :tokens, only: %i[index show create]
+      resources :analytics, only: [] do
+        collection do
+          scope :actions, module: 'analytics/actions' do
+            get :top_licenses_by_volume, path: 'top-licenses-by-volume', to: 'counts#top_licenses_by_volume'
+            get :top_urls_by_volume,     path: 'top-urls-by-volume',     to: 'counts#top_urls_by_volume'
+            get :top_ips_by_volume,      path: 'top-ips-by-volume',      to: 'counts#top_ips_by_volume'
+            get :count,                                                  to: 'counts#count'
+          end
         end
       end
 
-      resources :event_logs, path: 'event-logs', only: %i[index show]
+      post :search, to: 'searches#search'
     end
 
-    resources :metrics, only: %i[index show] do
-      collection do
-        scope :actions, module: 'metrics/actions' do
-          get :count, to: 'counts#count'
-        end
-      end
+    # Artifact :show action needs to be a bit loose with the Accept header, so we're
+    # defining the route outside of the restrictive mime type constraint above.
+    scope defaults: { format: :jsonapi } do
+      resources :release_artifacts, only: %i[show], path: 'artifacts', constraints: { id: /.*/, format: /.*/ }
     end
 
-    resources :analytics, only: [] do
-      collection do
-        scope :actions, module: 'analytics/actions' do
-          get :top_licenses_by_volume, path: 'top-licenses-by-volume', to: 'counts#top_licenses_by_volume'
-          get :top_urls_by_volume,     path: 'top-urls-by-volume',     to: 'counts#top_urls_by_volume'
-          get :top_ips_by_volume,      path: 'top-ips-by-volume',      to: 'counts#top_ips_by_volume'
-          get :count,                                                  to: 'counts#count'
-        end
+    # Release packages can support and respond with a variety of mime types, so
+    # we're defining those routes here with their own unique constraints.
+    namespace :release_packages, path: 'packages' do
+      scope :pypi, module: :pypi, constraints: MimeTypeConstraint.new(:html, raise_on_no_match: true), defaults: { format: :html } do
+        get 'simple/:id', to: 'simple#index', as: :pypi_packages
```

Comment:
We also need a `get 'simple'` route [according to PEP 503](https://peps.python.org/pep-0503/). So rename the current action to `#show` and create an `#index` listing all products with a `PYPI` distribution strategy.

Reply:
Note that all links to HTML pages, i.e. in the new `#index` action, should have a trailing slash according to the spec.

We can use `url_for(..., trailing_slash: true)`.